crates/fuzzing/src/generators/table_ops.rs

```diff
@@ -44,31 +48,50 @@ impl TableOps {
     /// calls), but is not guaranteed to avoid traps (might access out-of-bounds
     /// of the table).
     pub fn to_wat_string(&self) -> String {
-        let mut wat = "(module\n".to_string();
+        let mut module = Module::new();
 
         // Import the GC function.
-        wat.push_str("  (import \"\" \"gc\" (func))\n");
+        let mut imports = ImportSection::new();
+        imports.import("", "gc", ImportType::Function(0));
 
         // Define our table.
-        wat.push_str("  (table $table ");
-        write!(&mut wat, "{}", self.table_size()).unwrap();
-        wat.push_str(" externref)\n");
+        let mut tables = TableSection::new();
+        tables.table(TableType {
+            element_type: ValType::ExternRef,
+            limits: Limits {
+                min: self.table_size(),
+                max: None,
+            },
+        });
 
         // Define the "run" function export.
-        wat.push_str(r#"  (func (export "run") (param"#);
-        for _ in 0..self.num_params() {
-            wat.push_str(" externref");
+        let mut functions = FunctionSection::new();
+        functions.function(0);
```

Comment:
Since this is a type index I think you'll need to define the type with a `TypeSection` as well.

Reply:
Added! Though not sure if it's right.